packages/gluestick/src/config/webpack/webpack.config.client.js

```diff
@@ -50,14 +52,35 @@ module.exports = (
     };
   });
   config.plugins.push(
-    new chunksPlugin(
-      deepClone(configuration),
-      { silent: settings.silent, chunk_info_filename: settings.chunk_info_filename },
-    ),
     // Make it so *.server.js files return null in client
     new webpack.NormalModuleReplacementPlugin(/\.server(\.js)?$/, path.join(__dirname, './mocks/serverFileMock.js')),
     progressHandler(logger, 'client'),
     new DuplicatePackageChecker(),
   );
+
+  // If vendor Dll bundle exists, use it otherwise fallback to CommonsChunkPlugin.
+  const vendorDllManifestPath: string = path.join(
+    process.cwd(), gluestickConfig.buildDllPath, manifestFilename,
+  );
+  if (fs.existsSync(vendorDllManifestPath)) {
+    logger.info('Vendor DLL bundle found, using DllReferencePlugin');
+    config.plugins.unshift(
+      new webpack.DllReferencePlugin({
+        context: configuration.context,
+        manifest: require(vendorDllManifestPath),
+      }),
+    );
+    config.plugins.push(new ChunksPlugin(deepClone(configuration), { appendChunkInfo: true }));
+  } else {
+    logger.info('Vendor DLL bundle not found, using CommonsChunkPlugin');
+    config.plugins.push(
+      new webpack.optimize.CommonsChunkPlugin({
+        name: 'vendor',
+        filename: `vendor${process.env.NODE_ENV === 'production' ? '-[hash]' : ''}.bundle.js`,
+      }),
```

Comment:
This will cause problems in an internal project at TrueCar where we have several white label versions of the same app. Many of the partners have very similar chunks in the partner folder and using CommonsChunkPlugin groups them together. For that reason we can't rely on CommonsChunkPlugin to automatically generate the vendor bundle.

However, digging deeper into the webpack configuration it shows that you can use a `minChunks` function to isolate common chunks to the node_modules folder. Shown here: https://webpack.js.org/plugins/commons-chunk-plugin/#passing-the-minchunks-property-a-function

Lastly, even if we did use CommonsChunkPlugin and isolated it to the node_modules folder I have some concerns that a large app with many sub apps may not benefit as much from this. You could end up with a very large vendor bundle because 2 or 10 of the apps use a large dependency that the other 8 apps do not. Automating this with the CommonsChunkPlugin may work for many use cases but I'm not sure it will work for all use cases. Is it possible to make usage of the CommonsChunkPlugin optional. Instead it would be nice if you could specify a whitelist of vendor modules that will be included in vendor. Including CommonsChunkPlugin should not be included. Really it seems like the handling of how we build out the vendor bundle is a separate task than what we are trying to solve. Parallelizing builds and changing the way we split between apps and the vendor bundle are two separate issues.